openedx/frontend-app-course-authoring
# [FC-0036] feat: Add ContentTagsDrawer widget

## Description

This PR implements a side drawer widget for units' taxonomy tags. It includes displaying the Unit's tags, along with their lineage (ancestor tags) data. It also implements the listing the available taxonomy tags (including nesting ones) to select from to apply to this unit.

### Things to Note:
- The actual implementation for updating/edit the tags that belong to a unit will be implemented in the future PR.
- The way the nested taxonomy tags are fetched currently is a bit aggressive, makes a lot of requests, prefetching the data before it is needed/displayed. In interested of getting this PR out, I opted to update the implementation in a later PR, however if it a blocking issue in the review, I'll address it as part of this PR.

### Screenshots

![Screen Shot 2023-11-08 at 4 09 09 PM](https://github.com/openedx/frontend-app-course-authoring/assets/6829768/d4fa397c-bde6-4f59-a181-b4ed45126711)

![Screen Shot 2023-11-08 at 4 09 49 PM](https://github.com/openedx/frontend-app-course-authoring/assets/6829768/10f648ea-6741-4a7f-a46f-425674fe9d01)

![Screen Shot 2023-11-08 at 4 10 34 PM](https://github.com/openedx/frontend-app-course-authoring/assets/6829768/e1c68b2e-6865-410d-a261-412f782d267e)

## Supporting information

Related Tickets:
- https://github.com/openedx/modular-learning/issues/118

Related PRs:
- https://github.com/openedx/edx-platform/pull/33563

## Testing instructions

1. Make sure you followed the instructions mentioned in https://github.com/openedx/edx-platform/pull/33563 first
2. Check that the unit name correctly shows at the top of the drawer
3. Check that all the taxonomies are shown in the drawer
4. Check that a number badge is show next to taxonomies that have tags (the count represents the explicit tags only), if there are not tags applied, no badge should appear
5. Check that when you expand a section, it shows the tags that are applied to the unit, along with their ancestors if any (the x button currently does nothing, also part of a future PR)
6. Check that when clicking on "Add tags" button it a dropdown shows with the tags along with arrows if they have further nested tags. Clicking on the arrow, shows/hides the subtags. 
7. Check that you can click on the checkboxes (as of now, they do not do anything, just placeholders for a future PR)
8. Check that you can navigate the drawer using only the keyboard

---
Private-ref: [FAL-3523](https://tasks.opencraft.com/browse/FAL-3523)
